plugin/healthchecker/filter.go

```diff
@@ -0,0 +1,127 @@
+package healthchecker
+
+import (
+	"sync"
+	"time"
+
+	"github.com/hashicorp/golang-lru"
+	"github.com/miekg/dns"
+)
+
+type (
+	HealthCheckFilter struct {
+		cache    *lru.Cache
+		checker  Checker
+		interval time.Duration
+		names    map[string]struct{}
+	}
+
+	entry struct {
+		dnsRecord dns.RR
+		healthy   bool
+		quit      chan struct{}
+	}
+)
+
+var mtx sync.Mutex
```

Comment:
Can we move `mtx` into `HealthCheckFilter`?

Reply:
Then idk how to lock the mutex in evicted func.

Reply:
Create mutex in `NewHealthCheckFilter`, pass it to evict func, store to the HealthCheckFilter struct